src/evd/classic/evd.cpp

```diff
@@ -173,6 +173,85 @@ void evd_classic_tol(struct matrix_t Xmat, struct matrix_t Qmat, struct vector_t
     reorder_decomposition(evec, &Qmat, 1, greater);
     free(A);
 }
+void evd_cyclic(struct matrix_t Xmat, struct matrix_t Qmat, struct vector_t evec, double tol) {
```

Comment:
Can you please rename it as `evd_cyclic_tol` since I would be creating a different version without the tolerance parameter?